### ISSUE TITLE
[iOS] [a11y] Unable to fully navigate page using talkback when using nested scroll views

### ISSUE THREAD

**Issue by ChristianEdwardPadilla** [flutter/flutter#113615]
## Steps to Reproduce

1. Execute `flutter run` on the code sample
2. Enable iOS VoiceOver
3. Use VoiceOver to change focus from the top app bar to the scroll view elements.
4. Use VoiceOver to focus successive elements until the view port moves down the scroll view.
5. Use VoiceOver to focus previous elements, causing the view port to scroll back up.
6. Observe that there are elements at the top of the page that cannot be focused.

Expected:
VoiceOver should be able to access all elements on the screen. It should be able to tab through to the bottom and then tab back up to the top of the scroll view.

Actual:
VoiceOver is unable to return to elements at the initial top of the scroll view.

Code Sample:
```dart
import 'package:mobile.flutter.google_material.material2/google_material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: LongColumnContainer(),
    );
  }
}

class LongColumnContainer extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('A11y Repro App'),
      ),
      body: SingleChildScrollView(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            SizedBox(
              height: 16,
            ),
            Icon(
              Icons.favorite,
              size: 14,
            ),
            SizedBox(
              height: 16,
            ),
            Text(
              'Page title',
              textAlign: TextAlign.center,
            ),
            SizedBox(
              height: 16,
            ),
            Text(
              'Page subtitle',
              textAlign: TextAlign.center,
            ),
            SizedBox(
              height: 20,
            ),
            Text(
              'Title1',
            ),
            cardCarousel(),
            SizedBox(
              height: 20,
            ),
            Text(
              'Title2',
            ),
            SizedBox(height: 16),
            Text(
              'subtitle2',
            ),
            SizedBox(height: 16),
            cardCarousel(),
            SizedBox(
              height: 20,
            ),
            Icon(
              Icons.favorite,
              size: 100,
            ),
            SizedBox(
              height: 20,
            ),
            Text(
              'end of carousel',
              textAlign: TextAlign.center,
            ),
            Icon(
              Icons.favorite,
              size: 100,
            ),
            SizedBox(
              height: 20,
            ),
            Text(
              'end of page',
              textAlign: TextAlign.center,
            ),
            SizedBox(
              height: 20,
            ),
          ],
        ),
      ),
      drawer: Drawer(child: Text('drawer tile')),
    );
  }
}

Widget cardCarousel() {
  return SingleChildScrollView(
    scrollDirection: Axis.horizontal,
    child: IntrinsicHeight(
      child: Padding(
        padding: EdgeInsets.symmetric(horizontal: 14),
        child: Row(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            card(Icons.favorite, 'calendar', 'some small text'),
            SizedBox(width: 10),
          ],
        ),
      ),
    ),
  );
}

Widget card(IconData iconData, String title, String body) {
  return Semantics(
    container: true,
    explicitChildNodes: true,
    child: Container(
      width: 304,
      foregroundDecoration: BoxDecoration(
          borderRadius: BorderRadius.circular(8),
          border: Border.all(color: Colors.red)),
      child: ClipRRect(
        borderRadius: BorderRadius.circular(8),
        child: Column(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            SizedBox(
              height: 210,
              child: Icon(
                iconData,
                size: 200,
              ),
            ),
            Padding(
              padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 24),
              child: Column(
                mainAxisSize: MainAxisSize.min,
                crossAxisAlignment: CrossAxisAlignment.stretch,
                children: [
                  Text(
                    title,
                  ),
                  SizedBox(
                    height: 8,
                  ),
                  Text(
                    body,
                  ),
                ],
              ),
            ),
            Container(
              padding: const EdgeInsetsDirectional.fromSTEB(16, 0, 16, 24),
              alignment: Alignment.centerLeft,
              child: Text('View details'),
            ),
          ],
        ),
      ),
    ),
  );
}
```

flutter doctor:
```
[✓] Flutter (Channel google3, on Debian GNU/Linux rodete 5.18.16-1rodete1-amd64, locale
    en_US.UTF-8)
    • Framework revision fe9b598c12 (9 days ago), 2022-10-09T00:00:00.000
    • Engine revision ac1fec6367
    • Dart version 2f0c186d3b

[✓] Android toolchain - develop for Android devices (Android SDK version Stable)
    • Android SDK at google3
    • Platform Stable, build-tools Stable
    • Java binary at: /opt/android-studio-with-blaze-2021.3/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] Android Studio (version 4.2)
    • Android Studio at /opt/android-studio-with-blaze-4.2
    • Flutter plugin version 58.0.1
    • Dart plugin version 202.8443
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)

[✓] Android Studio (version 2020.3)
    • Android Studio at /opt/android-studio-with-blaze-2020.3
    • Flutter plugin version 61.2.2
    • Dart plugin version 203.8292
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] Android Studio (version 2021.3)
    • Android Studio at /opt/android-studio-with-blaze-2021.3
    • Flutter plugin version 69.0.3
    • Dart plugin version 213.7371
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2019.2)
    • IntelliJ at /opt/intellij-ce-stable

[✓] VS Code (version 1.69.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.15.1

[✓] Connected device (1 available)
    • Linux (desktop) • linux • linux-x64 • Debian GNU/Linux rodete 5.18.16-1rodete1-amd64

[✓] Google3 (on linux)
    • KVM enabled

• No issues found!
```

repro video attached:

https://user-images.githubusercontent.com/37954976/196496889-c911f183-e106-4af5-8868-a9381a36279f.mp4

**Comment by jacobsimionato** [comment]
Copied from internal bug b/211049563 which is considered urgent by a client

**Comment by sharvari-g** [comment]
This issue affects iOS voice over. 

Flutter doctor output-
```
[✓] Flutter (Channel google3, on macOS 12.6 21G115 darwin-x64, locale en)
    • Framework revision 1c8ae34942 (33 days ago), 2022-09-16T00:00:00.000
    • Engine revision 6610f3f2a9
    • Dart version 381de7ccbe

[✓] Android toolchain - develop for Android devices (Android SDK version Stable)
    • Android SDK at google3
    • Platform Stable, build-tools Stable
    • Java binary at: /Applications/Android Studio with Blaze Beta.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] GoogleMac (on macos)
    • AppleWWDRCA installed.
    • Google development certificate installed.
    • Device 00008020-000B059C1EC3002E is whitelisted

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio with Blaze Beta.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio with Blaze.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode_13.4.1.app/Contents/Developer
    • Build 13F100

[✓] Connected device (3 available)
    • Sharvari’s iPhone (mobile) • 00008020-000B059C1EC3002E            • ios        • iOS 15.6.1 19G82
    • iPhone 11 Pro (mobile)     • D4E435C2-30E8-4361-86B6-67D5462AA2ED • ios        • com.apple.CoreSimulator.SimRuntime.iOS-15-5 (simulator)
    • macOS (desktop)            • macos                                • darwin-x64 • macOS 12.6 21G115 darwin-x64


```

**Comment by ChristianEdwardPadilla** [comment]
Updated to reflect initial iOS report.

**Comment by ChristianEdwardPadilla** [comment]
Not able to repro on Android device (pixel 4a) or android emulator. So, this appears to be an iOS-specific a11y issue.

**Comment by goderbauer** [comment]
cc @chunhtai This is blocking a customer's release, see b/211049563

**Comment by chunhtai** [comment]
This seems to only happen when swipe fast, it seems to work correctly if wait for a few second before swipe back

**Comment by chunhtai** [comment]
It looks like this can be fixed by adding cache extent. The problem is that the page tile is too far away that it gets clipped from semantics tree(not sure why it does that for a single child scroll view). If swipe up too fast the scrollable doesn't have time to update the content so it think there is no more child to focus so it focuses the header instead.

I think there are two ways to fix this, increase cache extent or make singlechild scroll view to not clip the semantics child. I lean toward the later since it will be laid out anyway
